backend/forms/graphql/mutations/update_response_dimensions.py

```diff
@@ -0,0 +1,68 @@
+import graphene
+from graphene.types.generic import GenericScalar
+
+from access.cbac import graphql_check_access
+
+from ...models.field import Field, FieldType
+from ...models.survey import Survey
+from ...utils.process_form_data import process_form_data
+from ..response import FullResponseType
+
+
+class UpdateResponseDimensionsInput(graphene.InputObjectType):
+    event_slug = graphene.String(required=True)
+    survey_slug = graphene.String(required=True)
+    response_id = graphene.String(required=True)
+    form_data = GenericScalar(required=True)
+
+
+SENTINEL_BYPASS_PERMISSION_CHECK_FOR_TESTING = object()
+
+
+class UpdateResponseDimensions(graphene.Mutation):
+    class Arguments:
+        input = UpdateResponseDimensionsInput(required=True)
+
+    response = graphene.Field(FullResponseType)
+
+    @staticmethod
+    def mutate(
+        _root,
+        info,
+        input: UpdateResponseDimensionsInput,
+    ):
+        """
+        Called by the dimensions box submit button in
+        frontend/src/app/[locale]/events/[eventSlug]/surveys/[surveySlug]/responses/[responseId]/page.tsx
+
+        Each dimension can be either a SingleSelect or a MultiSelect, depending on whether multiple
+        values are allowed (or already associated).
+        """
+        form_data: dict[str, str] = input.form_data  # type: ignore
+
+        survey = Survey.objects.get(event__slug=input.event_slug, slug=input.survey_slug)
+        response = survey.responses.get(id=input.response_id)
+        dimensions = list(survey.dimensions.all())
+
+        # TODO bastardization of graphql_check_access, rethink
+        if info is not SENTINEL_BYPASS_PERMISSION_CHECK_FOR_TESTING:
```

Comment:
I'd probably opt to `mock.patch` `graphql_check_access` to do nothing instead of a sentinel like this.

Reply:
yeah… now that we're `SimpleNamespace`ing, let's also `SimpleNamespace` the `info` so that `info.context.user` doesn't throw